src/main/java/com/jcabi/github/mock/MkIssueLabels.java

```diff
@@ -154,20 +178,31 @@ public Label map(final XML xml) {
     public void remove(@NotNull(message = "name cannot be NULL")
         final String name
     ) throws IOException {
-        this.storage.apply(
-            new Directives().xpath(
-                String.format("%s/label[.='%s']", this.xpath(), name)
-            ).remove()
-        );
+        if (this.labelSet().contains(name)) {
+            this.storage.apply(
+                new Directives().xpath(
+                    String.format("%s/label[.='%s']", this.xpath(), name)
+                ).remove()
+            );
+            final MkIssueEvents events = new MkIssueEvents(
+                this.storage,
+                this.self,
+                this.repo
+            );
+            events.create(
+                Event.UNLABELED,
+                this.ticket,
+                this.self,
+                Optional.of(name)
+            );
+        }
     }
 
     @Override
     public void clear() throws IOException {
-        this.storage.apply(
-            new Directives().xpath(
-                String.format("%s/label", this.xpath())
-            ).remove()
-        );
+        for (final String label : this.labelSet()) {
+            this.remove(label);
```

Comment:
@cvrebert This doesn't do anything - it just removes elements from the set that `labelSet()` returns. Better use `iterator.remove()`.